src/applications/vaos/tests/mocks/fetch.js

```diff
@@ -0,0 +1,130 @@
+ * @param {boolean} [params.limit=false] Whether the mock should set the user as passing the request limit check
+ * @param {boolean} [params.requestPastVisits=false] Whether the mock should set the user as passing the past visits check
+ *    for requests
+ * @param {boolean} [params.directPastVisits=false] Whether the mock should set the user as passing the past visits check
+ *    for direct scheduling
+ * @param {Array<VARClinic|VAOSClinic>} [params.clinics=[]] The clinics returned during the eligibility checks
+ * @param {boolean} [params.pastClinics=false] Whether or not the mock should also mock an appointments fetch with an
+ *    past appointment with a clinic matching one passed in the clinics param, so that the user passes the past clinics check
+ * @param {number} [version=2] The version of the calls to set up, defaulted to version 2
+ * }
+ */
+export function mockEligibilityFetchesByVersion({
+  facilityId,
+  typeOfCareId,
+  siteId = null,
+  limit = false,
+  requestPastVisits = false,
+  directPastVisits = false,
+  clinics = [],
+  pastClinics = false,
+  version = 2,
+}) {
+  if (version === 2) {
+    setFetchJSONResponse(
+      global.fetch.withArgs(
+        `${
+          environment.API_URL
+        }/vaos/v2/patient?facility_id=${facilityId}&clinical_service_id=${typeOfCareId}&type=direct`,
+      ),
+      {
+        data: {
+          attributes: {
+            hasRequiredAppointmentHistory: directPastVisits,
+            isEligibleForNewAppointmentRequest: limit,
+          },
+        },
+      },
+    );
+    setFetchJSONResponse(
+      global.fetch.withArgs(
+        `${
+          environment.API_URL
+        }/vaos/v2/patient?facility_id=${facilityId}&clinical_service_id=${typeOfCareId}&type=request`,
+      ),
+      {
+        data: {
+          attributes: {
+            hasRequiredAppointmentHistory: requestPastVisits,
+            isEligibleForNewAppointmentRequest: limit,
+          },
+        },
+      },
+    );
+    setFetchJSONResponse(
+      global.fetch.withArgs(
+        `${
+          environment.API_URL
+        }/vaos/v2/locations/${facilityId}/clinics?clinical_service=${typeOfCareId}`,
+      ),
+      {
+        data: clinics,
+      },
+    );
+
+    const pastAppointments = clinics.map(clinic => {
+      const appointment = getVAAppointmentMock();
+      appointment.attributes = {
+        ...appointment.attributes,
+        startDate: moment().format(),
+        facilityId: facilityId.substr(0, 3),
+        sta6aid: facilityId,
+        clinicId: clinic.id,
+      };
+      appointment.attributes.vdsAppointments[0].currentStatus = 'FUTURE';
+
+      return appointment;
+    });
+    // These will probably need to in the different version blocks after we refactor
+    // how the past appointment check works
+    setFetchJSONResponse(
+      global.fetch.withArgs(
+        `${environment.API_URL}/vaos/v0/appointments?start_date=${moment()
```

Comment:
I'm going to make a ticket for refactoring our patient appt history code. It's not transformed or run through our services right now and we need to update it to be more in line with the rest of our service/domain logic.